### PR TITLE
In WrapperCategory: renaming UnderlyingCategory -> ModelingCategory

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

File: CAP/gap/WrapperCategory.gd [hunk 1]
```diff
@@ -6,6 +6,48 @@
...
+#! <HorLine/>
+#! <Row>
+#!   <Item>...</Item>
+#! </Row>
+#! <HorLine/>
+#! <Row>
+#!   <Item><C>cat_1 := CategoryConstructor_1( non_categorical_input )</C></Item>
+#! </Row>
+#! <HorLine/>
+#! </Table>
```
Comment (zickgraf) [inline CAP/gap/WrapperCategory.gd:45]: There might be towers with multiple roots, e.g. in the context of `FunctorCategories`. Of course that is harder to visualize, but I think it should be at least mentioned in the text.

File: CAP/gap/WrapperCategory.gd [hunk 2]
```diff
@@ -6,6 +6,48 @@
 
 #! @Chapter Create wrapper hulls of a category
 
+#! The support for towers of category constructors is one
+#! of the main design features of CAP. Many categories that
+#! appear in the various applications can be modeled by such towers.
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/WrapperCategory.gd:11]: changed to "can be modeled by towers of multiple category constructors"

File: CAP/gap/WrapperCategory.gd [hunk 3]
```diff
@@ -6,6 +6,48 @@
...
+#! </Row>
+#! <HorLine/>
+#! <Row>
+#!   <Item><C>cat_1 := CategoryConstructor_1( non_categorical_input )</C></Item>
+#! </Row>
+#! <HorLine/>
+#! </Table>
+#! The wrapper category <C>W</C> is by construction equivalent
+#! to the top category <C>cat_n</C> in the tower.
+#! In practice, the word <Q>tower</Q> stands for an atomic poset
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/WrapperCategory.gd:48]: atomic poset -> finite poset

**Comment by mohamed-barakat** [comment]
> 1. For deposited packages I usually provide backwards compatible deprecated operations (or at least an error message) because people out there might be using the old operations. Is this a policy we want to follow? If yes, please add backwards compatibility.

Done with your help, thanks.

> 2. I think the concept of towers should be explained at a more prominent point. But of course better here than nowhere, so this does not have to be part of this PR.

It has to be repeated here anyway. And yes, some of the sentences can be copied elsewhere to explain the philosophy of the categorical towers.
 
> PS: I think `DeductiveSystem` might have been broken since [6f463e1](https://github.com/homalg-project/CAP_project/commit/6f463e1533312f3447088172e4df1b8dbc9dd33f) :D

I see :D

**Pull request merged**